File: src/info/meodinger/LabelPlusFX/Util/CAccelerator.java
package info.meodinger.LabelPlusFX.Util;

import javafx.scene.input.GestureEvent;
import javafx.scene.input.KeyEvent;

/**
 * Author: Meodinger
 * Date: 2021/6/15
 * Location: info.meodinger.LabelPlusFX.Util
 */
public class CAccelerator {

    /**
     * Based on Windows Platform
     */

    private static final String OS = System.getProperty("os.name").toLowerCase();

    public static final boolean isMac = OS.contains("mac");

    public static boolean isControlDown(GestureEvent event) {
        if (event.isControlDown()) return true;
        if (isMac) return event.isMetaDown();
        return false;
    }

    public static boolean isControlDown(KeyEvent event) {
        if (event.isControlDown()) return true;
        if (isMac) return event.isMetaDown();
        return false;
    }

}

File: src/main/kotlin/info/meodinger/lpfx/util/CAccelerator.kt
package info.meodinger.lpfx.util.accelerator

import info.meodinger.lpfx.util.platform.isMac

import javafx.scene.input.GestureEvent
import javafx.scene.input.KeyEvent


/**
 * Author: Meodinger
 * Date: 2021/7/29
 * Location: info.meodinger.lpfx.util
 */

/**
 * Alias isMetaDown
 */
fun isControlDown(event: GestureEvent): Boolean {
    if (event.isControlDown) return true
    if (isMac && event.isMetaDown) return true

    return false
}

/**
 * Alias isMetaDown
 */
fun isControlDown(event: KeyEvent): Boolean {
    if (event.isControlDown) return true
    if (isMac && event.isMetaDown) return true

    return false
}

/**
 * Alias isMetaDown
 */
fun isAltDown(event: GestureEvent): Boolean {
    if (event.isAltDown) return true
    if (isMac && event.isMetaDown) return true

    return false
}

/**
 * Alias isMetaDown
 */
fun isAltDown(event: KeyEvent): Boolean {
    if (event.isAltDown) return true
    if (isMac && event.isMetaDown) return true

    return false
}